examples_tests/49.NewAPITest/main.cpp

```diff
@@ -0,0 +1,330 @@
+// Copyright (C) 2018-2020 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+#define _NBL_STATIC_LIB_
+#include <iostream>
+#include <cstdio>
+#include <nabla.h>
+#include <nbl/system/CWindowWin32.h>
+
+using namespace nbl;
+using namespace core;
+
+int main()
+{
+	constexpr uint32_t WIN_W = 800u;
+	constexpr uint32_t WIN_H = 600u;
+	constexpr uint32_t SC_IMG_COUNT = 3u;
+
+	const char* vs_source = R"(#version 430
+
+layout (location = 0) in vec2 Pos;
+layout (location = 1) in vec3 Color;
+
+layout (location = 0) out vec3 OutColor;
+
+void main()
+{
+	OutColor = Color;
+	gl_Position = vec4(Pos, 0.0, 1.0);
+}
+)";
+	const char* fs_source = R"(#version 430
+
+layout (location = 0) in vec3 InColor;
+layout (location = 0) out vec4 OutColor;
+
+void main()
+{
+	OutColor = vec4(InColor, 1.0);
+}
+)";
+
+	auto win = system::CWindowWin32::create(WIN_W, WIN_H, system::IWindow::ECF_NONE);
+
+	auto gl = video::IAPIConnection::create(video::EAT_OPENGL, 0, "New API Test");
+	auto surface = gl->createSurface(win.get());
```

Comment:
I'd much prefer
```cpp
auto gl = video::COpenGLConnection::create(0, "New API Test");
auto surface = video::COpenGLSurfaceWin32(win.get()); // maybe refcount the window?
```

Reply:
why? it would be great to not include backend headers in user code

Reply:
makes the library easier to extent with extension provided windows, and driver backends.